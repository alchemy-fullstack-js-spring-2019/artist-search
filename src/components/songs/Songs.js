import React from 'react';
import PropTypes from 'prop-types';
import Song from './Song';
import { Link } from 'react-router-dom';

function Songs({ songsArray, match }) {
  const songsList = songsArray.map((song, i) => {
    const songName = song.title;
    const songId = `/${match.params.artistName}/${match.params.artistId}/${match.params.albumId}/${songName}/${song.id}`;
    return (
      <li key={i}>
        <Link to={songId}><Song song={song.title} /></Link>
      </li>
    );
  });
  
  return (
    <ul>
      {songsList}
    </ul>
  );
}

Songs.propTypes = {
  songsArray: PropTypes.array.isRequired,
  match: PropTypes.object.isRequired
};

export default Songs;
 
