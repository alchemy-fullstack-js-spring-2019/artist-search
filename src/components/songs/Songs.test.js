import React from 'react';
import { shallow } from 'enzyme';
import Songs from './Songs';

describe('Songs component',  ()=>{
  it('renders stuff', ()=>{
    const wrapper = shallow(<Songs match={{ params: 'url' }} songsArray={[{ title: 'song2' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
