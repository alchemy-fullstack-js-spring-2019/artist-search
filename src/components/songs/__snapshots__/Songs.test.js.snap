// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Songs component renders stuff 1`] = `
<ul>
  <li
    key="0"
  >
    <Link
      to="/undefined/undefined/undefined/song2/undefined"
    >
      <Song
        song="song2"
      />
    </Link>
  </li>
</ul>
`;
